### PR TITLE
CI check the unused flag

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**
Added for check CI for unused functions.

### CODE REVIEW

File: feature-flag.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+flag:RAW_SCREEN_SERVER
+flag:MATRIX
```
Comment (RuffaloLavoisier) [inline feature-flag.md:2]: We have to let them know that these flags exist somewhere.

File: .github/buildEverything.py [hunk 2]
```diff
@@ -91,8 +91,13 @@ def doBuild(makeDebug):
 	
     ap.add_argument("-l", "--support-language", type=str, required=True, help="# model language to compile. (Enter 'all' to compile all language packs.)")
     ap.add_argument("-m", "--support-model", type=str, required=True, help="# model type to compile. (Enter 'all' to compile all model packs.)")
+    ap.add_argument("-f", "--support-feature", type=str, required=False, default="", help="# feature to compile. (Enter a feature to compile.)")
     args = vars(ap.parse_args())
 
+    if str(args["support_feature"]) != "":
+        logging.info('Setting flag ' + str(args["support_feature"]) + '...')
+        os.system('sed -i'' -r -e "/build_flags/a\    -D '+str(args["support_feature"])+'" platformio.ini')
```
Comment (RuffaloLavoisier) [inline .github/buildEverything.py:99]: convert `Win cmd`?

**Review by RuffaloLavoisier — COMMENTED** [review]
I think it needs to be corrected.

File: .github/buildEverything.py (hunk 2 above)
Comment (simonmicro) [reply .github/buildEverything.py:99]: Why did you ever use `os.system`? I think even here we used code to open the `platformio.ini` and to read it!

File: feature-flag.md (hunk 1 above)
Comment (RuffaloLavoisier) [reply feature-flag.md:2]: Plan B, there's also a way to add a flag [here](https://github.com/Open-Smartwatch/open-smartwatch.github.io/blob/3357dbb630603e2e33255d887dbf3b32512fa566/docs/resources/firmware.md?plain=1#L5). I think we can do it.In my opinion, this is the most ideal.

File: .github/workflows/test-FEATURE.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      uses: actions/setup-python@v2
+    - name: Install PlatformIO
+      run: python -m pip install --upgrade pip && pip install --upgrade platformio
+    
+    - name: Rename config
+      run: mv include/config.h.example include/config.h
+   
+    - name: Compile language ${{ matrix.language }} model ${{ matrix.model }} feature ${{ matrix.feature }}
+      run: python3 .github/buildEverything.py -l ${{ matrix.language }} -m ${{ matrix.model }} -f ${{ matrix.feature }}
+    - name: Upload firmware artifacts
```
Comment (simonmicro) [inline .github/workflows/test-FEATURE.yml:70]: I do not think we should archive the produced firmwares.

File: feature-flag.md (hunk 1 above)
Comment (simonmicro) [reply feature-flag.md:2]: Maybe it is used - but I think the choice of `Markdown` for this config is pretty bad. Maybe take a look at YAML / JSON instead?

**Comment by RuffaloLavoisier** [comment]
Current PR has resolved the current issue. I will open a new PR to update.

**Pull request merged**